### PR TITLE
docs: add plugin docs

### LINKED ISSUE #352: [DOC] add scanner plugin docs
### What kind of documentation improvement is needed?

None

### What is the change that is needed?

after #261 is merged, add docs for scanner plugin
context: https://github.com/project-copacetic/copacetic/pull/261#issuecomment-1746308127

### PR DESCRIPTION
**Pull request by sozercan**
Describe the changes in this pull request using active verbs such as _Add_, _Remove_, _Replace_ ...

Closes #352

### CODE REVIEW

File: website/docs/scanner-plugins.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+```
+
+# Community Scanner Plugins
+
+If you have built a scanner plugin and would like to add it to this list, please submit a PR to update this section with your plugin.
+
+:::tip
+
+If you have any issues with a specific plugin, please open an issue in the applicable plugin's repository.
+
```
Comment (anubhav06) [inline website/docs/scanner-plugins.md:28]: I'll open a PR to add [anubhav06/copa-grype](https://github.com/anubhav06/copa-grype) to the list, once this PR merges 👍🏻

**Review by salaxander — APPROVED** [review]

File: website/docs/scanner-plugins.md (hunk 1 above)
Comment (sozercan) [reply website/docs/scanner-plugins.md]: added, thank you!

File: website/docs/scanner-plugins.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+# Motivation
+
+By default, `copa` uses [Trivy](https://github.com/aquasecurity/trivy) to scan container images for vulnerabilities. However, we understand that different organizations have different requirements and may want to use different vulnerability scanners.
+
+Starting with v0.5.0 and later, `copa` offers extensibility to support different vulnerability scanners. Plugin architecture allows users to use the vulnerability scanner of their choice to patch container images without having to modify `copa`'s core codebase.
+
+# Usage
+
+Scanner plugin binaries must be in `$PATH` and should be prefixed with `copa-`. Copa will automatically detect and use the scanner plugin if it is in `$PATH`.
```
Comment (ritazh) [inline website/docs/scanner-plugins.md:13]: are there permissions required to execute the binary?

File: website/docs/scanner-plugins.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+# Scanner Plugin Interface
+
+Scanner plugins must implement the following interface:
+
+## v1alpha1
+
+```golang
+type UpdateManifest struct {
+    // API version of the interface (e.g. v1alpha1)
+    APIVersion string         `json:"apiVersion"`
```
Comment (ritazh) [inline website/docs/scanner-plugins.md:46]: should we add some verbiage about versioning, backward compatibility policy, etc?

File: website/docs/scanner-plugins.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    // Updates is a list of UpdatePackage that contains information about the package updates
+    Updates    UpdatePackages `json:"updates"`
+}
+
+// UpdatePackages is a list of UpdatePackage
+type UpdatePackages []UpdatePackage
+
+// Metadata contains information about the OS and config
+type Metadata struct {
+    OS     OS     `json:"os"`
```
Comment (ritazh) [inline website/docs/scanner-plugins.md:58]: does copa validate against a supported list of OS and Arch?

Comment (ritazh) [reply website/docs/scanner-plugins.md:58]: or do they non-supported ones just get skipped?

Comment (sozercan) [reply website/docs/scanner-plugins.md:58]: for OS, yes it'll validate and fail fast if it's not supported
for Arch, it's used for pulling the arch-specific image and vex output

File: website/docs/scanner-plugins.md (hunk 2 above)
Comment (sozercan) [reply website/docs/scanner-plugins.md]: added

File: website/docs/scanner-plugins.md (hunk 3 above)
Comment (sozercan) [reply website/docs/scanner-plugins.md:52]: added a note, let me know if you are looking for any specifics.

Comment (ritazh) [reply website/docs/scanner-plugins.md:52]: LGTM

**Review by ritazh — APPROVED** [review]
LGTM

**Pull request merged**